src/examples/grid-cells/fancy-input-grid-cell.js

```diff
@@ -50,16 +57,57 @@ class FancyInputGridCell extends React.Component {
   onKeyDown(event) {
     switch (event.key) {
       case 'Enter': {
-        this.setState(state => ({ interactive: !state.interactive }));
+        if (this.state.interactive) {
+          const nextCell = event.shiftKey ? this.props.minusY : this.props.plusY;
+
+          if (nextCell === this.props.gridCellRef) {
+            this.setState({
+              interactive: false,
+            });
+          } else {
+            this.setState({
+              interactive: false,
+            });
+            nextCell.current.focus();
+          }
+        } else {
+          this.setState({
+            interactive: true,
+          });
+        }
+
         break;
       }
       case 'Escape': {
-        this.setState({ interactive: false });
+        this.setState({
+          interactive: false,
+          wasEscaped: true,
+        });
         break;
       }
-      default: {
-        // Do not use keyboard navigation while interactive
+      case 'Tab': {
         if (this.state.interactive) {
+          const nextCell = event.shiftKey ? this.props.minusX : this.props.plusX;
+
+          if (nextCell === this.props.gridCellRef) {
+            this.setState({
+              interactive: false,
+            });
+          } else {
+            this.setState({
+              interactive: false,
+            });
+            nextCell.current.focus();
+          }
+
+          event.preventDefault();
+        }
+
+        break;
+      }
+      default: {
+        if (this.state.interactive && /Arrow/.test(event.key)) {
```

Comment:
Non-important nitpick:

Could make `/Arrow/` a const for this module and save a couple microseconds 😄

Reply:
Good point! At the moment, this is the "example grid cell" for fancy user interactions. I do want to extract valuable learnings into `GridCell` and `InteractiveGridCell` -- and I'll include the keyboard navigation `const`s (perhaps a `Map`? would be ideal to be able to augment keyboard interactions via the properties interface).